packages/core/contracts/financial-templates/optimistic-distributor/OptimisticDistributor.sol

```diff
@@ -0,0 +1,420 @@
+    uint256 public finalFee;
+
+    // Index of next created reward or proposal.
+    uint256 public nextCreatedReward;
+    uint256 public nextCreatedProposal;
+
+    // Rewards and proposals are mapped to their indices.
+    mapping(uint256 => Reward) public rewards;
+    mapping(uint256 => Proposal) public proposals;
+
+    // Immutable variables provided at deployment.
+    FinderInterface public immutable finder;
+    IERC20 public bondToken; // This cannot be declared immutable as bondToken needs to be checked against whitelist.
+
+    // Merkle Distributor can be set only once.
+    MerkleDistributor public merkleDistributor;
+
+    // Interface parameters that can be synced and stored in the contract.
+    StoreInterface public store;
+    OptimisticOracleInterface public optimisticOracle;
+
+    /********************************************
+     *                  EVENTS                  *
+     ********************************************/
+
+    event RewardCreated(
+        address indexed sponsor,
+        IERC20 rewardToken,
+        uint256 indexed rewardIndex,
+        uint256 maximumRewardAmount,
+        uint256 earliestProposalTimestamp,
+        uint256 optimisticOracleProposerBond,
+        uint256 optimisticOracleLivenessTime,
+        bytes32 priceIdentifier,
+        bytes customAncillaryData
+    );
+    event RewardIncreased(uint256 indexed rewardIndex, uint256 newMaximumRewardAmount);
+    event ProposalCreated(
+        address indexed sponsor,
+        IERC20 rewardToken,
+        uint256 indexed rewardIndex,
+        uint256 indexed proposalIndex,
+        uint256 proposalTimestamp,
+        uint256 maximumRewardAmount,
+        bytes32 merkleRoot,
+        string ipfsHash
+    );
+    event RewardDistributed(
+        address indexed sponsor,
+        IERC20 rewardToken,
+        uint256 indexed rewardIndex,
+        uint256 indexed proposalIndex,
+        uint256 maximumRewardAmount,
+        bytes32 merkleRoot,
+        string ipfsHash
+    );
+    event ProposalRejected(uint256 indexed rewardIndex, uint256 indexed proposalIndex);
+    event ProposalDeleted(uint256 indexed rewardIndex, uint256 indexed proposalIndex);
+    event MerkleDistributorSet(address indexed merkleDistributor);
+
+    /**
+     * @notice Constructor.
+     * @param _bondToken ERC20 token that the bond is paid in.
+     * @param _finder Finder to look up UMA contract addresses.
+     */
+    constructor(FinderInterface _finder, IERC20 _bondToken) {
+        finder = _finder;
+        require(_getCollateralWhitelist().isOnWhitelist(address(_bondToken)), "Bond token not supported");
+        bondToken = _bondToken;
+        syncUmaEcosystemParams();
+    }
+
+    /********************************************
+     *            FUNDING FUNCTIONS             *
+     ********************************************/
+
+    /**
+     * @notice Allows any caller to create a rewards object and deposit tokens that are linked to these rewards.
```

Comment:
throughout this contract you refer to it as a "rewards object". Consider changing this to a "Reward struct" which is more descriptive of the solidity data structure.

Reply:
fixed in df7e415f38dd6c6c8269693b2c59e42b288914f1